src/common/errors_js.ml

```diff
@@ -88,11 +88,106 @@ let print_reason_color ~first ~one_line ~color (message: message) =
   (if first then Printf.printf "\n");
   C.print ~color_mode:color to_print
 
-let print_error_color ~one_line ~color e =
-  let level, messages, trace_reasons = e in
-  let messages = append_trace_reasons messages trace_reasons in
-  print_reason_color ~first:true ~one_line ~color (List.hd messages);
-  List.iter (print_reason_color ~first:false ~one_line ~color) (List.tl messages)
+let file_location_style text = (C.Underline C.Default, text)
+let default_style text = (C.Normal C.Default, text)
+let source_fragment_style text = (C.Normal C.Default, text)
+let error_fragment_style text = (C.Normal C.Red, text)
+let line_number_style text = (C.Dim C.Default, text)
+
+(* TODO: Not sure how to handle the usecase when flow reads file content from stdin *)
+let load_file f =
+  let ic = open_in f in
+  let n = in_channel_length ic in
+  let s = Bytes.create n in
+  really_input ic s 0 n;
+  close_in ic;
+  (s)
+
+(* TODO: Add proper support for relative path, for example ../../.. *)
+let relative_path filename =
```

Comment:
Could you check that `src/common/utils_js.ml` or `src/common/files_js.ml` doesn't already have something you can reuse? We like to keep path manipulation code in one place.